### PR TITLE
Common artifacts for testing error scenarios for device and phoneNumber

### LINKED ISSUE #323: Artifact for test plan scenarios for device/phoneNumber errors
**Problem description**
In CAMARA we have guidelines for errors related to the management of device/phoneNumber inputs, and test plans should include scenarios to test that those guidelines are applied for APIs with such an input. 

Test guidelines give high level recommendations but they do not define specific scenarios, so different test definitions may end up with different scenarios for the same use case.

**Possible evolution**
Create a common artifact for these scenarios, here in Commonalities, that can be directly applied, or with some minor adaptation, to test plans. This would ease the creation of test plans and assure coherence. 

**Additional context**
This would be a first artifact but we may create additional artifacts for the testing of other functionalities that are shared by several APIs and have also common guidelines, such as:

* Subcriptions
* Events
* Common errors dealing with authentication/authorization

### PR DESCRIPTION
**Pull request by jlurien**
#### What type of PR is this?

* enhancement/feature
* tests

#### What this PR does / why we need it:

Adds the first common artifacts, with scenarios that can be included in test plans for operations that receive an input with device and phoneNumber. They must be aligned with the guidelines in Commonalities and indicate a proper way to validate that guidelines are enforced by the implementation.

Each artifact is assigned a code number, e.g. `C01`, and within each artifact, each scenario is numbered with the artifact code "." scenario number, e.g. `C01.01`, plus a short description. 

#### Which issue(s) this PR fixes:

Fixes #323 

#### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

#### Special notes for reviewers:

- The scenarios are intended to be aligned with guidelines being developed for forthcoming meta release Spring25, in particular:
  -  Error codes assume the new proposal being discussed in #321, to consolidate the error codes for both `device` and `phoneNumber`, and that breaks compatibility with the error codes used in past meta release Fall24. 
  -  Adds logic for new error `422 UNNECESSARY_IDENTIFIER`, which deprecated old use case for `403 INVALID_TOKEN_CONTEXT`. as proposed in PR #324

#### Changelog input

```
New common artifacts for test plans with operations that receive an input with `device` and `phoneNumber`. 

```

### CODE REVIEW

File: artifacts/testing/C01-device-errors.feature
Comment (mdomale) [inline artifacts/testing/C01-device-errors.feature:15]: We can have more clearer feature description to let know what should be removed or replaced  when using the templates. Mark the block to be removed when instantiating this feature file. Eg. $Hint_Start --to be removed { } $Hint_End .

File: artifacts/testing/C01-device-errors.feature
Comment (mdomale) [inline artifacts/testing/C01-device-errors.feature:18]: Scenario details are not clear,if we expect 422 then we anticipate 422 being considered and written in below mentioned scenario where we see only 400.

File: artifacts/testing/C01-device-errors.feature
Comment (mdomale) [inline artifacts/testing/C01-device-errors.feature:30]: same comment as above for this Scenario description as well.

File: artifacts/testing/C01-device-errors.feature
Comment (mdomale) [inline artifacts/testing/C01-device-errors.feature:44]: Mismatch of $.device.ipv4Address with value as /components/schemas/NetworkAccessIdentifier .Similarly for $.device.ipv6Address  & $.device.networkIdentifier parameters as well.

File: artifacts/testing/C01-device-errors.feature
Comment (jlurien) [reply artifacts/testing/C01-device-errors.feature:18]: If we decide to set such a priority (400 checks prior to 422 checks), then we can remove the comment.

Comment (jlurien) [reply artifacts/testing/C01-device-errors.feature:18]: I have the misleading comment, here it should be 400

File: artifacts/testing/C01-device-errors.feature
Comment (jlurien) [reply artifacts/testing/C01-device-errors.feature:44]: Fixed, thanks!

File: artifacts/testing/C01-device-errors.feature
Comment (jlurien) [reply artifacts/testing/C01-device-errors.feature:66]: Fixed, thanks

**Review by patrice-conil — DISMISSED** [review]
LGTM

**Review by PedroDiez — DISMISSED** [review]
LGTM

File: artifacts/testing/C01-device-errors.feature
Comment (mdomale) [reply artifacts/testing/C01-device-errors.feature:15]: @jlurien  Please clarify in more better way

**Comment by mdomale** [comment]
@jlurien  except above minor comments PR LGTM

File: artifacts/testing/C01-device-errors.feature
Comment (jlurien) [reply artifacts/testing/C01-device-errors.feature]: @mdomale, please review the new wording. Feel free to add any suggestions

**Comment by jlurien** [comment]
@mdomale @PedroDiez @patrice-conil hi, you may need to approve again after the last commit to address remaining comment

File: artifacts/testing/C02-phoneNumber-errors.feature
Comment (PedroDiez) [inline artifacts/testing/C02-phoneNumber-errors.feature:7]:
```suggestion
    * This is not a complete feature but a collection of scenarios that can be applied with minor

```

File: artifacts/testing/C01-device-errors.feature
Comment (PedroDiez) [inline artifacts/testing/C01-device-errors.feature:7]:
```suggestion
    * This is not a complete feature but a collection of scenarios that can be applied with minor

```

**Review by PedroDiez — APPROVED** [review]
Minor editorial

LGTM in advance

**Review by PedroDiez — APPROVED** [review]
LGTM

**Comment by mdomale** [comment]
LGTM

**Review by mdomale — APPROVED** [review]

**Review by patrice-conil — APPROVED** [review]
LGTM

**Comment by jlurien** [comment]
Hi @rartych, al past reviewers have given the LGTM. Please, double check if it is ready to be merged now. Thanks

**Review by rartych — APPROVED** [review]
LGTM

**Pull request merged**
